io/net/virt-net/lpm.py

```diff
@@ -88,45 +88,46 @@ def setUp(self):
         if not self.is_lpar_in_server(current_server, self.lpar):
             self.cancel("%s not in %s" % (self.lpar, current_server))
 
-        self.slot_num = str(self.params.get("slot_num", '*', default=1))
-        if int(self.slot_num) < 3 or int(self.slot_num) > 2999:
-            self.cancel("Slot invalid. Valid range: 3 - 2999")
-
-        self.bandwidth = str(self.params.get("bandwidth", default=2))
-
-        self.vios_name = self.params.get("vios_names", '*',
-                                         default=None).split(' ')
-        self.vios_id = []
-        for vios_name in self.vios_name:
-            self.vios_id.append(self.get_lpar_id(self.server, vios_name))
-        self.remote_vios_name = self.params.get("remote_vios_names", '*',
-                                                default=None).split(' ')
-        self.remote_vios_id = []
-        for vios_name in self.remote_vios_name:
-            self.remote_vios_id.append(self.get_lpar_id(self.remote_server,
-                                                        vios_name))
-
-        for vios in self.vios_id:
-            self.set_msp(self.server, vios)
-        for vios in self.remote_vios_id:
-            self.set_msp(self.remote_server, vios)
-
-        self.adapters = self.params.get("sriov_adapters",
-                                        default='').split(' ')
-        self.remote_adapters = self.params.get("remote_sriov_adapters",
-                                               default='').split(' ')
-        self.ports = self.params.get("sriov_ports",
-                                     default='').split(' ')
-        self.remote_ports = self.params.get("remote_sriov_ports",
+        if 'vnic' in self.options:
```

Comment:
Can't we have another way to determine if it is vnic based lpm or not. The options are not always expected to have the string vnic in them.

It would be better to have a yaml param to specify so.